From 2f7d61b2bfb59427926867c886595ff28dd50607 Mon Sep 17 00:00:00 2001
From: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Fri, 17 Dec 2021 15:50:17 -0300
Subject: core: Define DW_TAG_skeleton_unit if not available on current dwarf.h

We use this in both the dwarf_loader.c and in fprintf.c, so define it in
dwarves.h that is included in both.

Reported-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/all/YbkTAPn3EEu6BUYR@archlinux-ax161
Cc: Domenico Andreoli <domenico.andreoli@linux.com>
Cc: Douglas RAILLARD <douglas.raillard@arm.com>
Cc: Ilya Leoshkevich <iii@linux.ibm.com>
Cc: Jan Engelhardt <jengelh@inai.de>
Cc: Jiri Olsa <jolsa@kernel.org>
Cc: Matteo Croce <mcroce@microsoft.com>
Cc: Matthias Schwarzott <zzam@gentoo.org>
Cc: Yonghong Song <yhs@fb.com>
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 dwarves.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dwarves.h b/dwarves.h
index 52d162d..2132da0 100644
--- a/dwarves.h
+++ b/dwarves.h
@@ -1390,4 +1390,8 @@ extern bool no_bitfield_type_recode;
 
 extern const char tabs[];
 
+#ifndef DW_TAG_skeleton_unit
+#define DW_TAG_skeleton_unit 0x4a
+#endif
+
 #endif /* _DWARVES_H_ */
-- 
cgit 1.2.3-1.el7

